### ISSUE TITLE
Double wait period

### ISSUE THREAD

**Issue by bschulz1701** [NorthernWidget/MCP3421#5]
Commit e205bff81335e4e33a2a0fdd731aa90e6dd75141 added fixed delay times to the `GetVoltageRaw()` function 

This is unnecessary. This function already implemented polling for new result, the addition of the fixed wait times results in a double wait time - first the system polls for new result, then once it has that result, it waits the fixed time for a 'new' reading. 

**If it is desired to ensure a new data point is received, use calls `GetVoltageRaw(true)` or `GetVoltage(true)`, this will enforce the `WaitForVal` flag, and make sure the system waits for a new value.** 

It should also be noted that between the two, polling is a more efficient option as it will wait a _worst case_ of sample time (instead of a guarantee of sample time), and likely less (since some time will have passed since the last sample start if running in continuous mode) 

**This feature is slated for removal**

@ericbarefoot was there a particular use case this was designed to solve?

**Comment by ericbarefoot** [comment]
The issue I experienced using this feature was that even if a result was available, it was not at the required precision. Specifically, I was using the ADC to sample data from a laser rangefinder with analog output, but because the ADC has a much lower sampling rate at the full 18 bit resolution, the data resolution would be capped at higher sampling rates. 

That said, the modifications I made to Margay library to allow reading from the ADC use `MCP3421.GetVoltage()`, not `MCP3421.GetVoltage(true)`. The main use case for the ADC is in Margay, where the requirement ought to be to provide a reading at the specified resolution. The `WaitForVal` flag is not documented on Margay nor in the Margay library, and it is not documented clearly here either that the default is to not wait. 

It seems to me that a sensible solution would be to remove this feature here and make polling the default (`bool WaitForVal = true`), rather than the special case. I think this aligns better with the majority of use-cases. That said, it is not clear to me how the sensor would work in this scenario, since I have not tested it. 

Would that solution make sense to you, @bschulz1701 ? If so, if you would like to implement this, I can test here with a v2.2.
